src/core/stdcpp/utility.d

```diff
@@ -0,0 +1,43 @@
+/**
+ * D header file for interaction with Microsoft C++ <utility>
+ *
+ * Copyright: Copyright (c) 2018 D Language Foundation
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Manu Evans
+ * Source:    $(DRUNTIMESRC core/stdcpp/utility.d)
+ */
+
+module core.stdcpp.utility;
+
+import core.stdcpp.xutility : StdNamespace;
+
+extern(C++, (StdNamespace)):
+@nogc:
+
+/**
+* D language counterpart to C++ std::pair.
+*
+* C++ reference: $(LINK2 https://en.cppreference.com/w/cpp/utility/pair)
+*/
+struct pair(T1, T2)
+{
+    ///
+    alias first_type = T1;
+    ///
+    alias second_type = T2;
+
+    ///
+    T1 first;
+    ///
+    T2 second;
+
+    // FreeBSD has pair as non-POD so add a contructor
+    version (FreeBSD)
+    this(ref return scope inout(S) src) inout
```

Comment:
```suggestion
    this(ref return scope inout pair!(T1, T2) src) inout
```

Reply:
Does this change have any affect, or do you just prefer it?

Reply:
`S` isn't defined in the current scope:
```
/usr/home/braddr/sandbox/at-client/pull-4040470-FreeBSD_64_64/dmd/generated/freebsd/release/64/dmd -m64 -fPIC -w -I../../src -I../../import -Isrc -defaultlib= -debuglib= -dip1000 -L-lpthread -L-lm -L/usr/home/braddr/sandbox/at-client/pull-4040470-FreeBSD_64_64/druntime/generated/freebsd/release/64/libdruntime.a -O -release -extern-std=c++98 -main -unittest -version=CoreUnittest -c -of=./generated/freebsd/release/64/utility_d.o src/utility_test.d
../../src/core/stdcpp/utility.d(38): Error: undefined identifier `S`
```

Reply:
Yeah I just noticed that from the test failure ;)

Reply:
Welp, thats my fault for just blindly copying the language docs without thinking.